codalab/worker/bundle_service_client.py

```diff
@@ -22,27 +22,20 @@ def wrapper(*args, **kwargs):
                 raise BundleServiceException(message + ': ' + str(e), e.client_error)
             except urllib.error.HTTPError as e:
                 try:
-                    client_error = json.loads(e.read().decode())
-
-                    if client_error['error'] == 'invalid_grant':
+                    # Sets the file's current position at the offset 0. Otherwise, e.read() will return an empty string.
+                    e.seek(0)
```

Comment:
This part is a bit strange...the offset is not always at 0?

Reply:
There are two options:
1. Use `e.read()` but with `seek(0)` right before `e.read()`. Otherwise, `e.read()` will return an empty string from my experiments. 
2. Use `e.getvalue()`. This will directly return the real error message.
They are essentially the same.

Reply:
`e.getvalue()` seems much more principled / simpler than having to `seek` and `read`, so I'd go with that if possible.

Reply:
Actually, I just realized that I was wrong on this. `e.read()` without `e.seek(0)` should be the right way to handle it. The file pointer will originally be set at 0 when `e.read()`, but will be changed after `e.read()` get executed. When I was testing, I tried to print out the logging information by calling `e.read()` once, then assigned the value to `client_error`, so the pointer changed its position.

Reply:
If we do `x = e.read()` once, we can just use `x` and not depend on the cursor position or `e` for the error message, right?

Reply:
Yes, that's right. I have updated the code to use the standard `e.read()` in both files that have issues related to this.